### PR TITLE
IA-1973 - Update jupyter R image so user installed packages persist

### PR DESCRIPTION
**Pull request by gpcarr**
jupyter r image now installs image packages as root user and then creates directory `/home/jupyter-user/notebooks/packages` and uses that for user installed packages. 

removes directory `/home/jupyter-user/.rpackages` so `.libPaths()` doesn't have too many entries (we will be using new dir going forward since it's the same dir we mount the persistent disk on)

this PR also changes ownership of `/usr/local/lib/R/site-library` to `jupyter-user` as requested by @nturaga

### CODE REVIEW

File: terra-jupyter-r/Dockerfile [hunk 1]
```diff
@@ -161,9 +150,17 @@ RUN mkdir -p /home/jupyter-user/.rpackages \
...
+## After image installations, switch back to jupyter-user and create packages lib
+USER $USER
+RUN mkdir -p /home/jupyter-user/notebooks/packages \
+    && echo "R_LIBS=/home/jupyter-user/notebooks/packages" > /home/jupyter-user/.Renviron
 
 USER root
 
-RUN R -e 'IRkernel::installspec(user=FALSE)'
+RUN R -e 'IRkernel::installspec(user=FALSE)' \
+    && chown -R $USER:users /usr/local/lib/R/site-library
```
Comment (nturaga) [inline terra-jupyter-r/Dockerfile:164]: @gpcarr Do the pre-installed packages (devtools, remotes, etc..) go into `/usr/local/lib/R/site-library`?

Comment (gpcarr) [reply terra-jupyter-r/Dockerfile:164]: yep 
Screen Shot 2020-06-24 at 2 43 09 PM

**Comment by gpcarr** [comment]
- tested this by running automation tests in my FIAB using this docker image. 

- `NotebookRKernelSpec` had a few failures that were expected because I replaced `/home/jupyter-user/.rpackages` with `/home/jupyter-user/notebooks/packages`. When jenkins auto creates a leo PR to bump the `terra-jupyter-r` version, I'll fix these tests (just a simple string replacement). 

- I'll also push the leo code (in the gce-init script, and `jupyter-docker-compose-gce`) that makes user package installations persist to this auto created PR (adding the leo code will make package persistence fully functional). I have this code written already (since i have been using it to test) so i'll just cherry-pick that commit once the jenkins PR is set up.

- All tests in `NotebookBioconductorKernelSpec` pass as well (with no changes)

**Comment by rtitle** [comment]
I think the `terra-jupyter-r` changes look good as we've discussed. Do we also want to update the derived images `terra-jupyter-bioconductor`, `terra-jupyter-gatk`, `terra-jupyer-aou` to extend from the updated R image version? Or should that be in a separate PR?

**Comment by gpcarr** [comment]
> I think the `terra-jupyter-r` changes look good as we've discussed. Do we also want to update the derived images `terra-jupyter-bioconductor`, `terra-jupyter-gatk`, `terra-jupyer-aou` to extend from the updated R image version? Or should that be in a separate PR?

I didn't realize those extend `terra-jupyter-r`, I can definitely make those changes in this PR

**Comment by rtitle** [comment]
Sounds good, thanks

**Comment by Qi77Qi** [comment]
ah yea..derived images need to be updated as well

**Comment by gpcarr** [comment]
> lgtm once it's tested

I tested `terra-jupyter-r`, now I'm going to test the new `terra-jupyter-aou`, `terra-jupyter-gatk` and `terra-jupyter-bioconductor` by pushing a test image and running automation tests. will let you know when that's done but for now i bumped all those versions if you want to review that

**Comment by gpcarr** [comment]
@rtitle @Qi77Qi just tested aou, gatk and bioconductor images by running automation tests for them using my test `terra-jupyter-r` image, everything is passing

**Pull request merged**